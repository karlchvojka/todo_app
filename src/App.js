import React from 'react';
import './App.scss';

const App = () => {
  return (
    <div className="App">
      <header>
        <h1>React App Template</h1>
        </header>
      <main id="mainWrap">
        <section className="gridWrap">
          <h2>Welcome to The Grid.</h2>
          <p>This reactJS template was the result of a challenge from a close friend and mentor, and some curiosity. The purpose of this is to have an up and running ReactJS app, that is as minimal as possible.</p>
          <p>Proceed with caution. This may be a smaller app to start, but wont be as... 'comfy' as create-react-app.</p>
          <div className="theChallenge">
            <p>With that, I have one question...</p>
            <h4>What will YOU build today?</h4>
            </div>
          </section>
        </main>
      </div>
  )
}

export default App
