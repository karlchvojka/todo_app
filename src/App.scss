/*
  These styles are some defaults meant for the Template.
  These are default, feel free to change them.

  I would suggest leaving the imports tho ;)
*/

@import '../Styles/_reset.scss';
@import '../Styles/variables.scss';
@import '../Styles/fonts.scss';

body {
  background-color: $darkGrey;
  background-size: 75px 75px;
  background-image:
    linear-gradient(0deg, transparent 24%, rgba(255, 255, 255, .05) 25%, rgba(255, 255, 255, .05) 26%, transparent 27%, transparent 74%, rgba(255, 255, 255, .05) 75%, rgba(255, 255, 255, .05) 76%, transparent 77%, transparent),
    linear-gradient(90deg, transparent 24%, rgba(255, 255, 255, .05) 25%, rgba(255, 255, 255, .05) 26%, transparent 27%, transparent 74%, rgba(255, 255, 255, .05) 75%, rgba(255, 255, 255, .05) 76%, transparent 77%, transparent);
}

header {
  background-color: $headerGrey;
  padding:20px;
  text-align:center;
}
#mainWrap {
  height:100vh;
  max-width:80%;
  margin: 0 auto;

  .gridWrap {
    background-color: $headerGrey;
    border-radius: 10px;
    box-shadow: 2px 1px 8px 5px rgba(48,191,242,0.75);
    margin-top:50px;
    padding:$spacing;
    -webkit-box-shadow: 2px 1px 8px 5px rgba(48,191,242,0.75);
    -moz-box-shadow: 2px 1px 8px 5px rgba(48,191,242,0.75);

    h2 {
      text-align:center;
      margin-bottom: $spacing;
    }
    p {
      text-align: center;
    }

    .theChallenge {
      margin-top:20px;
      text-align: center;
    }
  }
}
