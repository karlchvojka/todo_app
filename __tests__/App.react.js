import React from 'react';
import {render} from '@testing-library/react'

import App from '../src/App.js';

it('should render properly', () => {
  const { asFragment } = render(<App />)
  expect (asFragment(<App />)).toMatchSnapshot()
});
