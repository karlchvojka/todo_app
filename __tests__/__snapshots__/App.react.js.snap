// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header>
      <h1>
        Todo App
      </h1>
    </header>
    <main
      id="mainWrap"
    >
      <section
        class="gridWrap"
      >
        <h2>
          The Burn.Net todo app
        </h2>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
